src/openassetio-core/include/openassetio/errors/exceptions.hpp

```diff
@@ -107,6 +108,12 @@ struct OPENASSETIO_CORE_EXPORT BatchElementException : OpenAssetIOException {
   BatchElementError error;
 };
 
+/// List of all OpenAssetIO-specific exceptions. Used in Python
+/// exception translation.
```

Comment:
[Consider][minor] As this is core, maybe mention languages more generically if the intention is to enable more programatic binding generation for any target language?

Reply:
Made more generic in b1373437